compiler/rustc_hir_typeck/src/fallback.rs

```diff
@@ -299,11 +295,13 @@ impl<'tcx> FnCtxt<'_, 'tcx> {
 
             let mut relationship = ty::FoundRelationships { self_in_trait: false, output: false };
 
-            for (vid, rel) in relationships.iter() {
-                if self.root_var(*vid) == root_vid {
-                    relationship.self_in_trait |= rel.self_in_trait;
-                    relationship.output |= rel.output;
-                }
+            let mut inner_infcx = self.infcx.inner.borrow_mut();
+            let relationships = inner_infcx.find_relationships(root_vid);
+            debug!("relationships: {:#?}", relationships);
+
+            for rel in relationships.iter() {
+                relationship.self_in_trait |= rel.self_in_trait;
+                relationship.output |= rel.output;
             }
```

Comment:
Should this loop be moved into `find_relationships`, which would just return a single `FoundRelationships` value?

Reply:
Done!